### PR TITLE
Fix Vikunja install script to prevent database deletion upon updating

### LINKED ISSUE #502: Vikunja update wont save data
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

After using the Vikunja update script the data is deleted.
It seems the folder `/opt/vikunja` is completed removed and the data is not saved somewhere else during the update.

### What settings are you currently utilizing?

- [ ] Default Settings
- [ ] Advanced Settings

### Which Linux distribution are you employing?

Debian 12

### If relevant, including screenshots or a code block can be helpful in clarifying the issue.

_No response_

### Please provide detailed steps to reproduce the issue.

Just run the update script and check for the already existing data.

### PR DESCRIPTION
**Pull request by vhsdream**
> [!NOTE]
> We are meticulous when it comes to merging code into the main branch, so please understand that we may reject pull requests that do not meet the project's standards. It's never personal. Also, game-related scripts have a lower chance of being merged.

## Description

This PR addresses an issue in which the Vikunja database (and files folder if created) gets deleted when the user updates Vikunja via the script. The change adds two more substitutions via the sed command to modify the default Vikunja config file; changes the location of the database file and the files folder to `/etc/vikunja`.

Fixes #502 

## Type of change
Please check the relevant option(s):

- [x] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)
- [x] Documentation updated (I have updated any relevant documentation) - **There is none, as far as I can tell**.

## Additional Information (optional)
Provide any additional context or screenshots about the feature or fix here.

## Related Pull Requests / Discussions

If there are other pull requests or discussions related to this change, please link them here:
- Related PR #

### CODE REVIEW

File: install/vikunja-install.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ cd /opt
 RELEASE=$(curl -s https://dl.vikunja.io/vikunja/ | grep -oP 'href="/vikunja/\K[0-9]+\.[0-9]+\.[0-9]+' | sort -V | tail -n 1)
 wget -q "https://dl.vikunja.io/vikunja/$RELEASE/vikunja-$RELEASE-amd64.deb"
 $STD dpkg -i vikunja-$RELEASE-amd64.deb
-sudo sed -i 's/^  timezone: .*/  timezone: UTC/' /etc/vikunja/config.yml
+sed -i 's|^  timezone: .*|  timezone: UTC|; s|"./vikunja.db"|"/etc/vikunja/vikunja.db"|; s|./files|/etc/vikunja/files|' /etc/vikunja/config.yml
```
Comment (havardthom) [inline install/vikunja-install.sh:29]: split to three lines for readability

**Review by havardthom — DISMISSED** [review]

**Review by MickLesk — DISMISSED** [review]

File: install/vikunja-install.sh (hunk 1 above)
Comment (havardthom) [reply install/vikunja-install.sh]: I did not mean split the string to three lines, please see other scripts how `sed` is commonly used. You can use `-e` for multiple expressions or separate `sed` commands. More than 2 expressions -> split to separate lines

Comment (vhsdream) [reply install/vikunja-install.sh]: Please let me know if the new change is sufficient.

**Review by havardthom — APPROVED** [review]

File: install/vikunja-install.sh (hunk 1 above)
Comment (MickLesk) [reply install/vikunja-install.sh]: i think @havardthom means like the tandoor-script (copy only)

```bash
sed -i -e "s|SECRET_KEY=.*|SECRET_KEY=$secret_key|g" \
       -e "s|POSTGRES_HOST=.*|POSTGRES_HOST=localhost|g" \
       -e "s|POSTGRES_PASSWORD=.*|POSTGRES_PASSWORD=$DB_PASS|g" \
       -e "s|POSTGRES_DB=.*|POSTGRES_DB=$DB_NAME|g" \
       -e "s|POSTGRES_USER=.*|POSTGRES_USER=$DB_USER|g" \
       -e "\$a\STATIC_URL=/staticfiles/" /opt/tandoor/.env
``` 

https://github.com/community-scripts/ProxmoxVE/blob/a4dd7f5449f4b1049b9a4bec2d8aec20286aab09/install/tandoor-install.sh#L68-L75

**Review by MickLesk — APPROVED** [review]

**Pull request merged**